### PR TITLE
Extra Metadata Error Handling

### PR DESCRIPTION
**Pull request by scvannost**
On branch CIDC-567
 Changes to be committed:
	modified:   cidc_api/resources/upload_jobs.py

### CODE REVIEW

File: cidc_api/resources/upload_jobs.py [hunk 1]
```diff
@@ -590,8 +590,14 @@ def extra_assay_metadata():
 
     try:
         UploadJobs.merge_extra_metadata(job_id, files)
+    except ValueError as e:
+        # thrown by parser itself if file cannot be parsed, e.g. wrong file uploaded
+        # wrapped by merger to include uuid / assay_hint information, just use that message
+        raise BadRequest(f"{e!s}\nIs it the wrong format or file?")
+
+    # TODO see if it's validation sort of error and return BadRequest
```
Comment (curlup) [inline cidc_api/resources/upload_jobs.py:598]: remove "TODO"?

**Review by curlup — APPROVED** [review]

**Comment by curlup** [comment]
make schemas dependency `>= 0.20.6`

File: requirements.modules.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@ flask-sqlalchemy~=2.4.0
 marshmallow-sqlalchemy==0.22.3
 google-cloud-storage~=1.31.0
 google-cloud-pubsub==0.42.1
-cidc-schemas~=0.20.3
+cidc-schemas>=0.20.6
```
Comment (curlup) [inline requirements.modules.txt:7]: `~=a.b.c` == `>=a.b.c && <a.b+1.*`
so leave as it's been 
```suggestion
cidc-schemas~=0.20.6
```

File: tests/resources/test_upload_jobs.py [hunk 3]
```diff
@@ -975,21 +975,35 @@ def test_extra_metadata(cidc_api, clean_db, monkeypatch):
...
-    merge_extra_metadata.assert_called()
+    assert res.status_code == 400
+    assert "files in request" in res.json["_error"]["message"]
+
+    # merge_extra_metadata = MagicMock()
+    # monkeypatch.setattr(
+    #     "cidc_api.models.UploadJobs.merge_extra_metadata", merge_extra_metadata
+    # )
+    # TODO add test that actually passes
+    # merge_extra_metadata.assert_called()
```
Comment (curlup) [inline tests/resources/test_upload_jobs.py:1006]: hm

**Review by curlup — COMMENTED** [review]
glanced only, will review tomorrow

File: tests/resources/test_upload_jobs.py (hunk 3 above)
Comment (curlup) [inline tests/resources/test_upload_jobs.py:1006]: yeah, so I'm only unsure about this here

**Review by curlup — CHANGES_REQUESTED** [review]

File: tests/resources/test_upload_jobs.py (hunk 3 above)
Comment (scvannost) [reply tests/resources/test_upload_jobs.py:1006]: I couldn't figure out how to make it run a test that would come back with a 200, I think I'll have to get/generate actual olink data

Comment (scvannost) [reply tests/resources/test_upload_jobs.py]: fixed by commit 3852882

**Review by jacoblurye — APPROVED** [review]
Nice work!

File: cidc_api/app.py [hunk 4]
```diff
@@ -43,6 +43,9 @@ def handle_errors(e: Exception):
             _error["message"] = e.data["messages"]
         else:
             _error["message"] = e.description
+
+        # general HTTP error log
+        print(f"HTTP {status_code}: {_error['message']}")
```
Comment (jacoblurye) [inline cidc_api/app.py:48]: 👍

File: tests/resources/test_upload_jobs.py [hunk 5]
```diff
@@ -975,21 +976,69 @@ def test_extra_metadata(cidc_api, clean_db, monkeypatch):
...
+        m.setattr(
+            "cidc_schemas.prism.merge_artifact_extra_metadata",
+            merge_artifact_extra_metadata,
+        )
+        res = client.post(
+            "/ingestion/extra-assay-metadata",
+            data={"job_id": job_id, "uuid-1": (io.BytesIO(b"fake file"), "fname1")},
+        )
+        if res.status_code != 200:
+            raise Exception(res.json["_error"]["message"])
```
Comment (jacoblurye) [inline tests/resources/test_upload_jobs.py:1012]:
```suggestion
```
Pretty sure this is left over from local debugging?

**Review by jacoblurye — APPROVED** [review]

**Review by curlup — APPROVED** [review]

**Pull request merged**